python/mxnet/pretrained_model.py

```diff
@@ -0,0 +1,363 @@
+
+        # Parse the file name to get prefix and epoch
+        _LOGGER.debug('param_file: %s' % param_file[0])
+        prefix = _os.path.splitext(param_file[0])[0]
+        epoch = prefix.split('-')[-1]
+        prefix = prefix[:-(len(epoch) + 1)]
+        prefix = _os.path.join(target_dir, prefix)
+        epoch = int(epoch)
+        _LOGGER.debug('prefix: %s, epoch: %s' % (prefix, epoch))
+
+        # Load the feedforward model
+        model = _FeedForward.load(prefix, epoch, ctx)
+
+        # Load the labels
+        label_file = _os.path.join(target_dir, 'labels.json')
+        _LOGGER.debug('label_file: %s' % label_file)
+        labels = _json.load(open(label_file))
+        return ImageClassifier(model, labels, metadata)
+    else:
+        raise TypeError('Unexpected model type: %s', metadata['model_type'])
+
+
+class ImageClassifier(object):
+    """
+    Wrapper of pretrained image classifier model.
+
+    Use :py:func:`load_model` or :py:func:`load_path` to load the model. Do not
+    construct directly.
+
+    Parameters
+    ----------
+    model :  FeedForward
+        The underlying model
+    labels : list
+        Map from index to label
+    metadata : dict
+        Metadata of the model including name, version, input shape, etc.
+    """
+    def __init__(self, model, labels, metadata):
+        self._model = model
+        self._labels = labels
+        self._input_shape = metadata['input_shape']
+        self.mean_nd = metadata['mean_nd']
+        self.mean_rgb = metadata['mean_rgb']
+        self.rescale = metadata['scale']
+        self._name = metadata['name']
+        self._version = metadata['version']
+
+    @property
+    def model(self):
+        return self._model
+
+    @property
+    def labels(self):
+        return self._labels
+
+    @property
+    def input_shape(self):
+        return self._input_shape
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def version(self):
+        return self._version
+
+    def __str__(self):
+        return "ImageClassifier: " + self._name + "(version %s)" % self._version
+
+    def __repr__(self):
+        return self.__str__()
+
+    def _make_dataiter(self, data, batch_size):
+        try:
+            import graphlab as _gl
+        except ImportError:
+            import sframe as _gl
+        except ImportError:
+            raise ImportError('Require GraphLab Create or SFrame')
+
+        if type(data) is not _gl.SFrame and type(data) is not _gl.SArray:
+            raise TypeError('Input data must be SFrame or SArray')
+        if type(data) is _gl.SArray and data.dtype() != _gl.Image:
+            raise TypeError('Expect image typed SArray, actual type is %s' % str(data.dtype()))
+        elif type(data) is _gl.SFrame:
+            if len(data.column_names()) != 1 or data.column_types()[0] != _gl.Image:
```

Comment:
It's probably a better user experience to allow multiple column SFrames, as long as there is only one image column.